### ISSUE TITLE
Guest order : Problem with the _old_xxx suffix

### ISSUE THREAD

**Issue by Zmove** [OFFLINE-GmbH/oc-mall-plugin#712]
Hello,

I'm playing with guest checkout, I noticed some problems :

The fact to rename the already existing email address by a random thing with _old_xxxx prevent the merchant to edit that user in the backend, cause when he will save the user, an error message will say that the email format in invalid.

The other problem is that it replace the email, but not the username, so if you want to create an account with the same email address, you will have an error saying that the username must be unique.

So, I suggest 2 things : 
- When you trigger the _old_xxx logic, apply it before the @ of the email address to keep a valid email address format
- Apply the same logic to the username column table

Best regards,

**Comment by tobias-kuendig** [comment]
Very good suggestions @Zmove. I absolutely see the need for this change.

**Comment by zlobec** [comment]
I see a miniature problem, when there are repeat sales to guest checkout from the same email... and old emails are renamed... When the admin wants to send a status change to the order... the error pops up:
```
Expected response code 354 but got code "503", with message "503-All RCPT commands were rejected with this error:
503-501 <@gmail.com_old_2021-03-04_151713>: malformed address: _old_2021-03-04_151713> may not follow <@gmail.com
503 Valid RCPT command must precede DATA
"
```
Would it be usefull to add a function to remove the _old_... before the email is sent?

**Comment by tobias-kuendig** [comment]
Good point @zlobec, the e-mail needs some clean up before it is sent :+1:

**Comment by tobias-kuendig** [comment]
Thanks again for your inputs @Zmove  and @zlobec.

This problem has been fixed in 7517a4f and 91323a8
